config/functions/common-functions

```diff
@@ -54,54 +54,126 @@ time_cal() {
 ## Install essential packages
 prepare_host() {
 
-	if [[ $(dpkg --print-architecture) != "amd64" ]]; then
-		error_msg "Your build host architecture is not supported,please use a 'x86-x64' build host."
-		return -1
-	fi
-
-	local build_host=$(lsb_release -sc)
-	info_msg "Build host: $build_host"
-
-	if [[ -z "$build_host" || "bionic focal" != *"$build_host"* ]]; then
-		error_msg "Your build host is not supported! Please use 'Ubuntu bionic(18.04)/focal(20.04)' as your build host or try to build in Docker."
-		return -1
-	fi
-
-	case $DISTRIB_RELEASE in
-		bionic)
-			if [ "$build_host" != "focal" -a "$build_host" != "bionic" ]; then
-				error_msg "Building Ubuntu Bionic(18.04) images requires a Ubuntu Bionic(18.04)/Focal(20.04) build host, please upgrade your build host or try to build in Docker."
-				return -1
-			fi
-		;;
-		buster|focal)
-			if [ "$build_host" != "focal" ]; then
-				error_msg "Building Debian Buster(10)/Ubuntu Focal(20.04) images requires a Ubuntu Focal(20.04) build host, please upgrade your build host or try to build in Docker."
-				return -1
-			fi
-		;;
+	HOST_ARCH=$(dpkg --print-architecture)
```

Comment:
add a check before running dpkg to see if the distro is apt based, if not, dump the package list to the console so that the user can DIY the preparation, or conveniently use the container,